### PR TITLE
Refactor local printing

### PR DESCRIPTION
**Pull request by bjorn3**
This fixes a panic when trying to print a struct. It shows `<struct>` for now until I implement real support for structs.

### CODE REVIEW

File: src/symbol.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 //! This module provides a naive implementation of symbolication for the time being.
 //! It should be expanded to support multiple data sources.
 
+#[macro_use]
```
Comment (nbaksalyar) [inline src/symbol.rs:4]: I think we can follow the 2018 edition style here, so we can replace this with
```rust
use dwarf_utils::dwarf_attr;
```

File: src/symbol/dwarf_utils.rs [hunk 2]
```diff
@@ -1,7 +1,94 @@
-use gimli::{DebuggingInformationEntry, Dwarf, Unit};
+use gimli::{DebuggingInformationEntry, Dwarf, Unit, UnitOffset, ValueType};
 
 use super::Reader;
 
+macro_rules! dwarf_attr {
+    (str($dwarf:ident,$unit:ident) $entry:ident.$name:ident || $missing:ident) => {
+        if let Some(attr) = $entry.attr(gimli::$name)? {
+            dwarf_attr_not_missing_action!($missing, $dwarf.attr_string(&$unit, attr.value())?)
```
Comment (nbaksalyar) [inline src/symbol/dwarf_utils.rs:8]: Nitpick - I'd suggest using a more descriptive name, e.g.
```suggestion
            dwarf_attr_exists_action!($missing, $dwarf.attr_string(&$unit, attr.value())?)
```

**Review by nbaksalyar — COMMENTED** [review]
It looks _so_ much better with `EvalContext`!
I left a couple of suggestions & nitpicks about style/naming.

File: src/symbol.rs (hunk 1 above)
Comment (bjorn3) [reply src/symbol.rs:4]: As far as I know that only works for extern crates and the unstable decl macro 2.0.

**Comment by bjorn3** [comment]
When a review suggests only small changes can I merge after resolving them and having all tests pass or do you want to review again @nbaksalyar?

**Review by nbaksalyar — APPROVED** [review]
Sure! When there are no major changes after a review, please feel free to merge.

Thank you :+1:

**Pull request merged**